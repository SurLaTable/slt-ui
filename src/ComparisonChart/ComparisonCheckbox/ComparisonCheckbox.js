import React from 'react';
import { connect } from 'react-redux';

import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
import MuiThemeProvider from '@material-ui/core/styles/MuiThemeProvider';
import Checkbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';

import { actionToggleProductSelection } from '../actions/productComparisonActions';

const theme = createMuiTheme({
	typography: {
		fontSize: 22
	}
});

let ComparisonCheckbox = (props) => {
	return (
		<MuiThemeProvider theme={theme}>
			<div className="comparison-checkbox">
				<FormControlLabel
					control={
						<Checkbox
							onChange={(event, checked) => {
								props.dispatch(
									actionToggleProductSelection(
										props.product,
										checked
									)
								);
							}}
							color="#333333"
							disabled={props.disabled}
							checked={props.checked}
							data-product={props.product}
						/>
					}
					label={props.checked ? 'Added' : 'Compare'}
				/>
			</div>
		</MuiThemeProvider>
	);
};

ComparisonCheckbox = connect((state, props) => {
	console.log('state:', state.productComparisonReducer);
	let selection = state.productComparisonReducer
		? state.productComparisonReducer.selection
		: [];
	let checked = false;

	// TODO: Figure out why `selection` is populated with `undefined`.
	if (selection.length) {
		for (let i = 0; i < selection.length; i++) {
			// console.log(selection);
			if (selection[i].id === props.product) {
				checked = true;
				break;
			}
		}
	}

	let disabled = !checked && selection.length > 2;
	return {
		...props,
		checked,
		disabled
	};
})(ComparisonCheckbox);

export default ComparisonCheckbox;
