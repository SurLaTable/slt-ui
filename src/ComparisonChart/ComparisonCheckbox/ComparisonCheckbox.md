```jsx
<StoreProvider>
	<ComparisonCheckbox product="PRO-13603" />
	<ComparisonCheckbox product="PRO-1006931" />
</StoreProvider>
```

```html
<div data-component="StoreProvider">
	<div data-component="ComparisonCheckbox" data-prop-product="PRO-13603" />
	<div data-component="ComparisonCheckbox" data-prop-product="PRO-1006931" />
</div>
```
