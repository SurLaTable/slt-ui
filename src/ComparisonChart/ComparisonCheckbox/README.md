```jsx
<StoreProvider>
	<ComparisonCheckbox product="PRO-13603" />
</StoreProvider>
```

React Habitat Code:

```html
est
<div data-component="StoreProvider">
	<div data-component="ComparisonCheckbox" data-prop-product="PRO-13603" />
</div>
```
