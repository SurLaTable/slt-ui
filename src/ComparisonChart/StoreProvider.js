import React from 'react';
// import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import sltReducers from './reducers/sltReducers';

const sltStore = createStore(
    sltReducers,
    // Enable DevTools, switch to localStorage on Prod:
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

const StoreProvider = (props) => (
    <Provider store={sltStore}>
			<React.Fragment>{props.children}</React.Fragment>
    </Provider>
);
export default StoreProvider;
