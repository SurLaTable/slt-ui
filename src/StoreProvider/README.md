```
<div data-component="StoreProvider"></div>
```
