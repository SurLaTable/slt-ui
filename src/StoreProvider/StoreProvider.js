import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import configureStore from '../utils/storeTools';

export const store = configureStore();

const StoreProvider = (props) => (
	<Provider store={store}>
		<React.Fragment>{props.children}</React.Fragment>
	</Provider>
);

StoreProvider.propTypes = {
	children: PropTypes.node
};

export default StoreProvider;
