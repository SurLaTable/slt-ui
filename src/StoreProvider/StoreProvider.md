```jsx
<StoreProvider />
```

```html
<div data-component="StoreProvider"></div>
```
