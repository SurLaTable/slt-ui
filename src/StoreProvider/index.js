import React from 'react';
import { Provider } from 'react-redux';
import configureStore from '../utils/storeTools';

export const store = configureStore();
const StoreProvider = (props) => (
	<Provider store={store}>
		<React.Fragment>{props.children}</React.Fragment>
	</Provider>
);
export default StoreProvider;
