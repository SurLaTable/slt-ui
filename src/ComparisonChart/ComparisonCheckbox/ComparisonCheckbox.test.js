import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ComparisonCheckbox from './ComparisonCheckbox';
import StoreProvider from '../../StoreProvider';

test('Render', () => {
	const renderer = new ShallowRenderer();
	renderer.render(
		<StoreProvider>
			<ComparisonCheckbox />
		</StoreProvider>
	);
	const result = renderer.getRenderOutput();
	expect(result).toMatchSnapshot();
});
