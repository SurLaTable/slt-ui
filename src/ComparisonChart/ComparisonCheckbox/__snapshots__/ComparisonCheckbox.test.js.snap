// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render 1`] = `
<Provider
  store={
    Object {
      "asyncReducers": Object {
        "productComparisonReducer": [Function],
      },
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <React.Fragment>
    <Connect(ComparisonCheckbox) />
  </React.Fragment>
</Provider>
`;
